libs/@local/hash-graph-types/graph_types/_schema.py

```diff
@@ -13,13 +14,19 @@
 from pydantic.json_schema import JsonSchemaValue
 from pydantic_core import CoreSchema, core_schema
 
+if typing.TYPE_CHECKING:
+    from . import GraphApiProtocol
+
+G = TypeVar("G", bound="GraphApiProtocol")
```

Comment:
If you're using a protocol, you don't need a TypeVar; you can use the class name as the type hint.

Both approaches are correct and valid.